tests/unit/CertificateGenerationService.unitTest.ts

```diff
@@ -593,6 +593,25 @@ describe("Certificate Generation Service", () => {
     });
   });
 
+  describe("welsh address logic", () => {
+    context("test STOP_WELSH_GEN EV variable", () => {
+      it("should circumvent the Welsh certificate generation logic if set to true", () => {
+        process.env.STOP_WELSH_GEN = "TRUE";
+        const logSpy = jest.spyOn(console, "log");
+        const certGenSvc = new CertificateGenerationService(
+            null as any,
+            new LambdaService(new Lambda())
+        );
+        certGenSvc.generateCertificate(mockTestResult)
+            .catch(() => {
+              expect(logSpy).toHaveBeenCalledWith(
+                  "Welsh certificate generation deactivated via environment variable set to TRUE"
```

Comment:
Why is this being tested with a catch block?

Reply:
I am calling the generateCertificate method directly with only the mockTestResult, for this test I am only interested in whether or not that log message is displayed to validate the IF logic has worked.

Without the catch-block the test passes but will throw an 'UnhandledPromiseRejectionWarning error' for all subsequent logic that continues to run but obviously doesn't have the config/context etc. To avoid having to mock all all subsequent logic, the catch is a way for me to prove the IF logic is correct and handle the warning message.